sycl/include/CL/sycl/backend_types.hpp

```diff
@@ -23,7 +23,8 @@ enum class backend : char {
   opencl = 1,
   level_zero = 2,
   cuda = 3,
-  all = 4
+  all = 5,
+  rocm = 4,
```

Comment:
Sorry, I wasn't clear enough last time. When compiled, enums are replaced with the constants. Imagine user compiles their code with compiler before your patch. `backend::all` equals to 4. Now they update their system with a newer compiler installation. The value in user binary is still 4. But for SYCL runtime library 4 now means `builtin::rocm`. This will lead to very strange behavior for user application. We'd like to avoid such confusion, so `all` must remain 4, and `rocm` must be assigned with 5.

```suggestion
  all = 4,
  rocm = 5,
```